ScoutSuite/providers/azure/facade/aad.py

```diff
@@ -1,68 +1,98 @@
-from azure.graphrbac import GraphRbacManagementClient
+import os
+
+from msgraphcore import GraphSession
 
 from ScoutSuite.core.console import print_exception
-from ScoutSuite.providers.utils import run_concurrently
-from ScoutSuite.utils import get_user_agent
 
 
 class AADFacade:
 
     def __init__(self, credentials):
         self.credentials = credentials
 
-    def get_client(self):
-        client = GraphRbacManagementClient(self.credentials.get_credentials('aad_graph'),
-                                         tenant_id=self.credentials.get_tenant_id())
-        client._client.config.add_user_agent(get_user_agent())
-        return client
+
+    async def _get_microsoft_graph_response(self, api_resource, api_version='v1.0'):
+        scopes = ['https://graph.microsoft.com/.default']
+
+        client = GraphSession(self.credentials.get_credentials(), scopes)
+        endpoint = 'https://graph.microsoft.com/{}/{}'.format(api_version, api_resource)
+
+        try:
+            response = client.get(endpoint)
+            if response.status_code == 200:
+                return response.json()
+            else:
+                print_exception('Failed to query Microsoft Graph endpoint \"{}\": status code {}'.
+                                format(api_resource, response.status_code))
+                return {}
+        except Exception as e:
+            print_exception('Failed to query Microsoft Graph endpoint \"{}\": {}'.format(api_resource, e))
+            return {}
 
     async def get_users(self):
         try:
-            # This filters down the users which are pulled from the directory, otherwise for large tenants this
```

Comment:
Is this still being handled?

Reply:
If you are taking about the filter of the user_type for guest then yes I am still handling it.
users_filtered = [d for d in users if d['userType'] in 'Guest']

Reply:
Right, but the difference is that in the old implementation the request did the filtering, whereas now you're making it client-side. So if there are e.g. 20k users in the tenant, you'll have to receive all of that information and filter locally.

Reply:
Fix now I filter in the get request with a filter that looks like this : '?$filter=userType+eq+%27Guest%27'